File: RootTools2/src/main/java/com/rarnu/tools/neo/xposed/ads/FuckCleanMaster.java
package com.rarnu.tools.neo.xposed.ads;

import com.rarnu.tools.neo.xposed.XpUtils;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.callbacks.XC_InitPackageResources;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

import java.util.Map;

/**
 * Created by rarnu on 11/18/16.
 */
public class FuckCleanMaster {

    public static void fuckCleanMaster(XC_LoadPackage.LoadPackageParam loadPackageParam) {
        XpUtils.findAndHookMethod("com.miui.optimizecenter.result.DataModel", loadPackageParam.classLoader, "post", Map.class, XC_MethodReplacement.returnConstant(""));
        XpUtils.findAndHookMethod("com.miui.optimizecenter.config.MiStat", loadPackageParam.classLoader, "getChannel", XC_MethodReplacement.returnConstant("international"));
        Class<?> clsAdImageView = XpUtils.findClass(loadPackageParam.classLoader, "com.miui.optimizecenter.widget.AdImageView");
        Class<?> clsAdvertisement = XpUtils.findClass(loadPackageParam.classLoader, "com.miui.optimizecenter.result.Advertisement");
        if (clsAdImageView != null && clsAdvertisement != null) {
            XpUtils.findAndHookMethod("com.miui.optimizecenter.result.CleanResultActivity", loadPackageParam.classLoader, "startAdCountdown", clsAdImageView, clsAdvertisement, XC_MethodReplacement.returnConstant(null));
            XpUtils.findAndHookMethod("com.miui.optimizecenter.result.CleanResultActivity", loadPackageParam.classLoader, "addAdvertisementEvent", String.class, clsAdvertisement, XC_MethodReplacement.returnConstant(null));
        }
    }

    public static void fuckResource(XC_InitPackageResources.InitPackageResourcesParam initPackageResourcesParam) {
        initPackageResourcesParam.res.setReplacement(initPackageResourcesParam.packageName, "string", "no_network", "");
    }

}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/xposed/ads/FuckCleanMaster.kt
@file:Suppress("Duplicates")

package com.rarnu.tools.neo.xposed.ads

import android.content.Context
import android.content.Intent
import android.preference.PreferenceManager
import com.rarnu.xfunc.*

/**
 * Created by rarnu on 11/18/16.
 */
object FuckCleanMaster {

    fun fuckCleanMaster(pkg: XposedPkg) {
        pkg.findClass("com.miui.optimizecenter.result.DataModel").findMethod("post", Map::class.java).hook { replace { result = "" } }
        pkg.findClass("com.miui.optimizecenter.config.MiStat").findMethod("getChannel").hook { replace { result = "international" } }

        val clsAdImageView = pkg.findClass("com.miui.optimizecenter.widget.AdImageView")
        val clsAdvertisement = pkg.findClass("com.miui.optimizecenter.result.Advertisement")
        if (clsAdImageView != null && clsAdvertisement != null) {
            pkg.findClass("com.miui.optimizecenter.result.CleanResultActivity").apply {
                findMethod("startAdCountdown", clsAdImageView, clsAdvertisement).hook { replace { result = null } }
                findMethod("addAdvertisementEvent", String::class.java, clsAdvertisement).hook { replace { result = null } }
            }
        }
        pkg.findClass("com.miui.optimizecenter.Application").findMethod("attachBaseContext", Context::class.java).hook {
            after {
                val pref = PreferenceManager.getDefaultSharedPreferences(thisObject as Context)
                pref.edit().putBoolean("key_information_setting_close", false).apply()
            }
        }
    }

    fun fuckSecurityCenter(pkg: XposedPkg) {
        pkg.findClass("com.miui.securitycenter.Application").findMethod("attachBaseContext", Context::class.java).hook {
            after {
                val pref = PreferenceManager.getDefaultSharedPreferences(thisObject as Context)
                pref.edit().putBoolean("key_information_setting_close", false).apply()
            }
        }
        pkg.findClass("com.miui.securitycenter.service.i").apply {
            findMethod("oI").hook { replace { result = null } }
            findMethod("oy").hook { replace { result = null } }
            findMethod("a", String::class.java, Intent::class.java).hook { replace { result = null } }
        }
    }

    fun fuckResource(res: XposedRes) {
        res.res.setReplacement(res.packageName, "string", "no_network", "")
    }

}
